src/compiler/gcc.rs

```diff
@@ -157,7 +157,7 @@ counted_array!(pub static ARGS: [ArgInfo<ArgData>; _] = [
     take_arg!("-aux-info", OsString, Separated, PassThrough),
     take_arg!("-b", OsString, Separated, PassThrough),
     flag!("-c", DoCompilation),
-    take_arg!("-dependency-file", PathBuf, Separated, PreprocessorArgumentPath),
+    take_arg!("-dependency-file", PathBuf, Separated, DepArgumentPath),
```

Comment:
Why is it not in clang.rs in the first place?

Reply:
I don't know; on the GCC side, it is apparently an actual option, though it appears to have a completely different meaning:

https://gcc.gnu.org/onlinedocs/gcc-9.3.0/gcc/Darwin-Options.html#index-dependency-file

Reply:
I'll move it as part of this PR; I need to tweak the title anyway.